find.go

```diff
@@ -53,6 +53,109 @@ func (s *server) findMultihashSubtree(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *server) findMetadata(w http.ResponseWriter, r *http.Request) {
+	var rb []byte
+	discardBody(r)
+	if r.Method != http.MethodGet {
+		http.Error(w, "", http.StatusNotFound)
+		return
+	}
+
+	rcode, resp := s.doFindMetadata(r.Context(), r.Method, findMethodOrig, r.URL, rb)
+	if rcode != http.StatusOK {
+		http.Error(w, "", rcode)
+		return
+	}
+	httpserver.WriteJsonResponse(w, http.StatusOK, resp)
+}
+
+func (s *server) doFindMetadata(ctx context.Context, method, source string, req *url.URL, body []byte) (int, []byte) {
+	start := time.Now()
+	latencyTags := []tag.Mutator{tag.Insert(metrics.Method, method)}
+	loadTags := []tag.Mutator{tag.Insert(metrics.Method, source)}
+	defer func() {
+		_ = stats.RecordWithOptions(context.Background(),
+			stats.WithTags(latencyTags...),
+			stats.WithMeasurements(metrics.FindLatency.M(float64(time.Since(start).Milliseconds()))))
+		_ = stats.RecordWithOptions(context.Background(),
+			stats.WithTags(loadTags...),
+			stats.WithMeasurements(metrics.FindLoad.M(1)))
+	}()
+
+	sg := &scatterGather[*url.URL, []byte]{
+		targets: s.servers,
+		tcb:     s.serverCallers,
+		maxWait: config.Server.ResultMaxWait,
+	}
+
+	var count int32
+	// TODO: wait for the first successfull response instead
+	if err := sg.scatter(ctx, func(cctx context.Context, b *url.URL) (*[]byte, error) {
+		// Copy the URL from original request and override host/schema to point
+		// to the server.
+		endpoint := *req
+		endpoint.Host = b.Host
+		endpoint.Scheme = b.Scheme
+		log := log.With("backend", endpoint)
+
+		bodyReader := bytes.NewReader(body)
+
+		req, err := http.NewRequestWithContext(cctx, method, endpoint.String(), bodyReader)
+		if err != nil {
+			log.Warnw("Failed to construct find-metadata backend query", "err", err)
+			return nil, err
+		}
+		req.Header.Set("X-Forwarded-Host", req.Host)
+		resp, err := s.Client.Do(req)
+		if err != nil {
+			log.Warnw("Failed to query backend for metadata", "err", err)
+			return nil, err
+		}
+		defer resp.Body.Close()
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			log.Warnw("Failed to read find-metadata backend response", "err", err)
+			return nil, err
+		}
+
+		switch resp.StatusCode {
+		case http.StatusOK:
+			atomic.AddInt32(&count, 1)
+			return &data, nil
+		case http.StatusNotFound:
+			atomic.AddInt32(&count, 1)
+			return nil, nil
+		default:
+			return nil, fmt.Errorf("status %d response from backend %s", resp.StatusCode, b.String())
+		}
+	}); err != nil {
+		log.Errorw("Failed to scatter HTTP find metadata request", "err", err)
+		return http.StatusInternalServerError, nil
+	}
+
+	var res []byte
+	for md := range sg.gather(ctx) {
+		if len(md) == 0 {
+			continue
+		}
+		// we need just first successfull response, no need to gather all
```

Comment:
* "successful" is consistently misspelled in this file. While at it please do a find/replace for it.

* I recommend documenting why finding the first entry is enough: key to metadata is unique in the context of provider+contextID etc.

```suggestion
		// we need just first successful response, no need to gather all
```